app/routes/session.py

```diff
@@ -139,11 +141,15 @@ def login() -> Response:
 
 
 def _set_questionnaire_supplementary_data(
-    questionnaire_store: QuestionnaireStore, metadata: MetadataProxy
+    *,
+    questionnaire_store: QuestionnaireStore,
+    metadata: MetadataProxy,
+    schema: QuestionnaireSchema,
 ) -> None:
     """
     If the survey metadata has an sds dataset id, and it either doesn't match what it stored, or there is no stored supplementary data
     then fetch it and add it to the store
+    This includes verification that the supplementary data lists cover any schema dependent lists which aren't populated by a list collector
```

Comment:
Do we want to update this doc string now you removed that `schema_dependent_lists` var?

Reply:
Yeah good idea - have amended that comment and shortened it a bit

Reply:
Happy this is resolved.